src/redux/constructor/constructor.operations.js

```diff
@@ -0,0 +1,430 @@
+import { gql } from '@apollo/client';
+import { client } from '../../utils/client';
+import { getFromLocalStorage } from '../../services/local-storage.service';
+import { constructorTranslations } from '../../translations/constructor.translations';
+
+export const deleteConstructorBasic = async (payload) => {
+  const token = getFromLocalStorage('HORONDI_AUTH_TOKEN');
+  const result = await client.mutate({
+    variables: {id:payload.constructorElementID},
+    context: { headers: { token } },
+    mutation: gql`
+        mutation($id: ID!) {
+            deleteConstructorBasic(id: $id) {
+                ... on ConstructorBasic {
+                    _id
+                }
+                ... on Error {
+                    message
+                    statusCode
+                }
+            }
+        }
+    `,
+    fetchPolicy: 'no-cache'
+  });
+  await client.resetStore();
+  const { data } = result;
+
+  if (data.deleteConstructorBasic.message) {
+    throw new Error(
+      `${data.deleteConstructorBasic.statusCode} ${
+        constructorTranslations[data.deleteConstructorBasic.message]
+      }`
+    );
+  }
+  return data.deleteConstructorBasic;
+};
+
+export const createConstructorBasic = async (payload) => {
+  const token = getFromLocalStorage('HORONDI_AUTH_TOKEN');
+  const result = await client.mutate({
+    context: { headers: { token } },
+    variables: {constructorElement:payload.constructorElement},
+    mutation: gql`
+        mutation($constructorElement: ConstructorBasicInput!) {
+            addConstructorBasic(constructorElement: $constructorElement) {
+                ... on ConstructorBasic {
+                    _id
+                    available
+                    basePrice{
+                        value
+                        currency
+                    }
+                    name {
+                        value
+                        lang
+                    }
+                    image
+                    material{
+                        _id
+                        name {
+                            value
+                            lang
+                        }
+                        color{
+                            colorHex
+                        }
+                    }
+                }
```

Comment:
This request repeated several times
Why don't you create separate variable with it

Reply:
Fixed